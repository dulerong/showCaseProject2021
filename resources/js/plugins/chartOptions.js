import Vue from 'vue'

export const chartOptionsLine = (axisTitle, chartTitle) => {
  return {
    chart: {
      type: 'line',
      zoom: {
        enabled: false
      }
    },
    markers: {
      size: 5
    },
    dataLabels: {
      enabled: false
    },
    stroke: {
      curve: 'straight'
    },
    title: {
      text: `${chartTitle}`,
      align: 'left'
    },
    legend: {
      position: 'left'
    },
    grid: {
      xaxis: {
        lines: {
          show: true
        }
      },   
      yaxis: {
        lines: {
          show: true
        }
      },
    },
    xaxis: {
      categories: ['8月', '9月', '10月', '11月', '12月', '1月', '2月', '3月', '4月', '5月', '6月', '7月'],
    },
    yaxis: {
      opposite: true,
      title: {
        text: axisTitle
      }
    },
  }
}
Vue.prototype.$_chartOptionsLine = chartOptionsLine

export const chartOptionsBar = (axisTitle, chartTitle) => {
  return {
    chart: {
      type: 'bar',
      height: 350
    },
    plotOptions: {
      bar: {
        horizontal: false,
        columnWidth: '55%',
        endingShape: 'rounded'
      },
    },
    dataLabels: {
      enabled: false
    },
    stroke: {
      show: true,
      width: 2,
      colors: ['transparent']
    },
    title: {
      text: `${chartTitle}`,
      align: 'left'
    },
    xaxis: {
      categories: ['8月', '9月', '10月', '11月', '12月', '1月', '2月', '3月', '4月', '5月', '6月', '7月'],
    },
    yaxis: {
      title: {
        text: axisTitle
      }
    },
    fill: {
      opacity: 1
    },
    tooltip: {
      y: {
        formatter: function (val) {
          return val
        }
      }
    },
    legend: {
      position: 'top',
      horizontalAlign: 'left',
    }
  }
}
Vue.prototype.$_chartOptionsBar = chartOptionsBar

export const chartOptionsStackedBar = (axisTitle, chartTitle, xAxisData) => {
  return {
    chart: {
      type: 'bar',
      height: 350,
      stacked: true,
    },
    colors: ['#2E93fA', '#66DA26', '#546E7A', '#E91E63', '#FF9800', '#E91E63', '#E91E63', '#E91E63', '#E91E63', '#E91E63', '#E91E63', '#E91E63'],
    plotOptions: {
      bar: {
        horizontal: true,
      },
    },
    stroke: {
      width: 1,
      colors: ['#fff']
    },
    title: {
      text: `${chartTitle}`,
      align: 'left'
    },
    xaxis: {
      categories: xAxisData,
      title: {
        text: axisTitle
      },
      labels: {
        formatter: function (val) {
          return val
        }
      }
    },
    yaxis: {
      title: {
        text: undefined
      },
    },
    tooltip: {
      y: {
        formatter: function (val) {
          return val
        }
      }
    },
    fill: {
      opacity: 1
    },
    legend: {
      position: 'top',
      horizontalAlign: 'left',
      offsetX: 40
    }
  }
}
Vue.prototype.$_chartOptionsStackedBar= chartOptionsStackedBar

export const chartSelector = (type, axisTitle, chartTitle, xAxisData) => {
  switch(type) {
    case '折線':
      return chartOptionsLine(axisTitle, chartTitle)
    case '棒':
      return chartOptionsBar(axisTitle, chartTitle)
    case '積上':
      return chartOptionsStackedBar(axisTitle, chartTitle, xAxisData)
  }
}
Vue.prototype.$_chartSelector = chartSelector