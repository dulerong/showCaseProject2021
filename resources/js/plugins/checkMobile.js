import Vue from 'vue'

const MOBILE_REGEX = [
  /Android/i,
  /webOS/i,
  /iPhone/i,
  /iPad/i,
  /iPod/i,
  /BlackBerry/i,
  /Windows Phone/i
]

const IS_MOBILE = () => MOBILE_REGEX.some(item => navigator.userAgent.match(item))

const IS_TOUCH_SCREEN = () => navigator.maxTouchPoints > 0 || IS_MOBILE()
// iOS 12 or below has no navigator.maxTouchPoints, hence need to check userAgent

Vue.prototype.$_IS_TOUCH_SCREEN = IS_TOUCH_SCREEN()