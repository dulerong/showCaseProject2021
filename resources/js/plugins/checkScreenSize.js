import Vue from 'vue'

export const screenWidthUnder450 = () => screen.width < 450

Vue.prototype.$_screenWidthUnder450 = screenWidthUnder450()