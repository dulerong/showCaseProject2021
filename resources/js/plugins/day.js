import Vue from 'vue'
import dayjs from 'dayjs'
import ja from 'dayjs/locale/ja'
dayjs.locale(ja)

const dateToday = dayjs()

Vue.prototype.$_dateToday = dateToday

Vue.prototype.$_dateTodayYYYYMMDD = dateToday.format('YYYY-MM-DD')

Vue.prototype.$_time_right_now = dateToday.format('YYYY-MM-DD HH:mm:ss')

Vue.prototype.$_hour_right_now = dateToday.format('HH')