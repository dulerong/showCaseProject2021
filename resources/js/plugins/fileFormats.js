import Vue from 'vue'

const MAX_UPLOAD_SIZE = 5 * 1000 * 1000 // byte * 1000 (kb) * 1000 (mb) = X megabyte
Vue.prototype.$_MAX_UPLOAD_SIZE = MAX_UPLOAD_SIZE

const ALLOWED_FILE_FORMATS = {
  csv: 'text/csv',
  jpg: 'image/jpeg',
  jpeg: 'image/jpeg',
  png: 'image/png',
  doc: 'application/msword',
  docx: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
  pdf: 'application/pdf',
  // ppt: 'application/vnd.ms-powerpoint',
  // pptx: 'application/vnd.openxmlformats-officedocument.presentationml.presentation',
  xls: 'application/vnd.ms-excel',
  xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
}
Vue.prototype.$_ALLOWED_FILE_FORMATS = ALLOWED_FILE_FORMATS

const findFileExtension = (fileName) => {
  const index = fileName.lastIndexOf('.')
  return fileName.slice(index + 1).toLowerCase()
}
Vue.prototype.$_findFileExtension = findFileExtension

const checkAllowedFileFormat = (fileExt, fileType) => {
  return Object.keys(ALLOWED_FILE_FORMATS).includes(fileExt) ?
    fileType === ALLOWED_FILE_FORMATS[fileExt] :
    false
}
Vue.prototype.$_checkAllowedFileFormat = checkAllowedFileFormat