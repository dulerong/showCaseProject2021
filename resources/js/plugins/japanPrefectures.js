import Vue from 'vue'

export const prefectures = [
  'hokkaido',
  'aomori',
  'iwate',
  'miyagi',
  'akita',
  'yamagata',
  'fukushima',
  'ibaragi',
  'tochigi',
  'gunma',
  'saitama',
  'chiba',
  'tokyo',
  'kanagawa',
  'niigata',
  'toyama',
  'ishikawa',
  'fukui',
  'yamanashi',
  'nagano',
  'gifu',
  'shizuoka',
  'aichi',
  'mie',
  'shiga',
  'kyoto',
  'osaka',
  'hyogo',
  'nara',
  'wakayama',
  'tottori',
  'shimane',
  'okayama',
  'hiroshima',
  'yamaguchi',
  'tokushima',
  'kagawa',
  'ehime',
  'kochi',
  'fukuoka',
  'saga',
  'nagasaki',
  'kumamoto',
  'oita',
  'miyazaki',
  'kagoshima',
  'okinawa'
]

Vue.prototype.$_prefectures = prefectures.map((item, i) => ({ name: item, code: i+1 }))