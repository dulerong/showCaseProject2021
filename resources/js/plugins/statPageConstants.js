import Vue from 'vue'

const statSites = [
  { value: '新北市板橋區', name: 'banqiao' },
  { value: '新北市三重區', name: 'sanchong' },
  { value: '新北市中和區', name: 'zhonghe' },
  { value: '新北市永和區', name: 'yonghe' },
  { value: '新北市新莊區', name: 'xinzhuang' },
  { value: '新北市新店區', name: 'xindian' },
  { value: '新北市土城區', name: 'tucheng' },
  { value: '新北市汐止區', name: 'xizhi' },
  { value: '新北市樹林區', name: 'shuling' },
  { value: '新北市鶯歌區', name: 'yingge' },
  { value: '新北市三峽區', name: 'sanxia' },
  { value: '新北市淡水區', name: 'danshui' },
  { value: '新北市瑞芳區', name: 'ruifang' },
]
Vue.prototype.$_statSites = statSites

const statMotherAges = [
  { value: '未滿15歲', name: 'ageUnder15', order: 1 },
  { value: '15～19歲', name: 'age15To19', order: 2 },
  { value: '20～24歲', name: 'age20To24', order: 3 },
  { value: '25～29歲', name: 'age25To29', order: 4 },
  { value: '30～34歲', name: 'age30To34', order: 5 },
  { value: '35～39歲', name: 'age35To39', order: 6 },
  { value: '40～44歲', name: 'age40To44', order: 7 },
  { value: '45～49歲', name: 'age45To49', order: 8 },
  { value: '50歲以上', name: 'age50Above', order: 9 }
]
Vue.prototype.$_statMotherAges = statMotherAges

const statBirthOrders = [
  { value: '1胎', name: 'birth1' },
  { value: '2胎', name: 'birth2' },
  { value: '3胎', name: 'birth3' },
  { value: '4胎', name: 'birth4' },
  { value: '5胎', name: 'birth5' },
  { value: '6胎', name: 'birth6' },
  { value: '7胎', name: 'birth7' },
  { value: '8胎', name: 'birth8' },
  { value: '9胎以上', name: 'birth9' }
]
Vue.prototype.$_statBirthOrders = statBirthOrders

const statSex = [
  { value: '男', name: 'male' },
  { value: '女', name: 'female' }
]
Vue.prototype.$_statSex = statSex

export const filterStatData = (data, siteSelected, birthOrderSelected, motherAgeSelected, sexSelected) => {
  return data
    .filter(item => siteSelected && siteSelected.length > 0 ? siteSelected.includes(item.site_id) : true)
    .filter(item => birthOrderSelected && birthOrderSelected.length > 0 ? birthOrderSelected.includes(item.birth_order) : true)
    .filter(item => motherAgeSelected && motherAgeSelected.length > 0 ?  motherAgeSelected.includes(item.mother_age) : true)
    .filter(item => sexSelected && sexSelected.length > 0 ? sexSelected.includes(item.birth_sex) : true)
    .filter(item => item.birth_count > 0)
}
Vue.prototype.$_filterStatData = filterStatData

export const sortMotherAges = (a, b) => {
  a = statMotherAges.find(item => item.value === a).order
  b = statMotherAges.find(item => item.value === b).order

  return a - b
}
Vue.prototype.$_sortMotherAges = sortMotherAges

export const translateData = (data) => {
  return data
    .map(({statistic_yyy, according, ...item}) => item)
    .map(item => ({
      ...item,
      site_id: translateDisctrict(item.site_id),
      mother_age: translateMotherAges(item.mother_age),
      birth_order: translateBirthOrders(item.birth_order),
      birth_sex: translateBirthSex(item.birth_sex)
    }))
}
Vue.prototype.$_translateData = translateData

export const translateDisctrict = (site) => {
  switch(site) {
    case '新北市板橋區':
      return 'banqiao'
    case '新北市三重區':
      return 'sanchong'
    case '新北市中和區':
      return 'zhonghe'
    case '新北市永和區':
      return 'yonghe'
    case '新北市新莊區':
      return 'xinzhuang'
    case '新北市新店區':
      return 'xindian'
    case '新北市土城區':
      return 'tucheng'
    case '新北市汐止區':
      return 'xizhi'
    case '新北市樹林區':
      return 'shuling'
    case '新北市鶯歌區':
      return 'yingge'
    case '新北市三峽區':
      return 'sanxia'
    case '新北市淡水區':
      return 'danshui'
    case '新北市瑞芳區':
      return 'ruifang'
  }
}

export const translateMotherAges = (age) => {
  switch(age) {
    case '未滿15歲':
      return 'ageUnder15'
    case '15～19歲':
      return 'age15To19'
    case '20～24歲':
      return 'age20To24'
    case '25～29歲':
      return 'age25To29'
    case '30～34歲':
      return 'age30To34'
    case '35～39歲':
      return 'age35To39'
    case '40～44歲':
      return 'age40To44'
    case '45～49歲':
      return 'age45To49'
    case '50歲以上':
      return 'age50Above'
  }
}

export const translateBirthOrders = (order) => {
  switch(order) {
    case '1胎':
      return 'birth1'
    case '2胎':
      return 'birth2'
    case '3胎':
      return 'birth3'
    case '4胎':
      return 'birth4'
    case '5胎':
      return 'birth5'
    case '6胎':
      return 'birth6'
    case '7胎':
      return 'birth7'
    case '8胎':
      return 'birth8'
    case '9胎以上':
      return 'birth9'
  }
}

export const translateBirthSex = (sex) => {
  switch(sex) {
    case '男':
      return 'male'
    case '女':
      return 'female'
  }
}