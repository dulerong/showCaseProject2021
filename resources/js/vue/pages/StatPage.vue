<template lang="pug">
div
  h1 New Taipei City Birth Statistics (2019)
  v-form(v-model="isValid")
    v-row(no-gutters)
        v-select(label="Site" :items="sites" v-model="siteSelected" :rules="rulesSelect" item-text="name" item-value="value" multiple chips clearable)
    v-row(no-gutters)
        v-select(label="Mother Age" :items="motherAges" v-model="motherAgeSelected" :rules="rulesSelect" multiple chips clearable)
    v-row(no-gutters)
        v-select(label="Birth Order" :items="birthOrders" v-model="birthOrderSelected" :rules="rulesSelect" multiple chips clearable)
    v-row(no-gutters)
      div(style="width: 150px") 
        v-select(label="Birth Sex" :items="sex" v-model="sexSelected" :rules="rulesSelect" multiple chips clearable)
    v-row(no-gutters justify="space-between" justify-lg="start")
      v-btn(@click="fetchData" color="success" :disabled="!isValid" :small="$_screenWidthUnder450" :class="{ 'mr-2' : !$_screenWidthUnder450 }") Fetch Data
      v-btn(@click="resetFilter" color="info" :small="$_screenWidthUnder450" :class="{ 'mx-2' : !$_screenWidthUnder450 }") Reset Filter
      v-btn(@click="clearData(true)" color="warning" :small="$_screenWidthUnder450" :class="{ 'mx-2' : !$_screenWidthUnder450 }") Clear Data
  v-row(no-gutters class="my-3")
    h3 Result: {{ totalNumberCount }} {{ totalNumberCount ? 'born' : '' }}
  v-data-table(:headers="headers" :items="data")
    template(v-slot:item.birth_count="{ item }")
      v-chip(:color="getColor(item.birth_count)" dark) {{ item.birth_count }}
</template>

<script>
import mockData from '~js/mockData/stat'
import { mapMutations } from 'vuex'
export default {
  data: (vm) => ({
    isValid: false,
    sites: vm.$_statSites,
    motherAges: vm.$_statMotherAges,
    birthOrders: vm.$_statBirthOrders,
    sex: vm.$_statSex,
    sexSelected: null,
    siteSelected: null,
    motherAgeSelected: null,
    birthOrderSelected: null,
    rulesSelect: [],
    headers: [
      { text: 'Site', align: 'start', sortable: false, value: 'site_id', },
      { text: 'Birth Order', value: 'birth_order' },
      { text: 'Birth Sex', value: 'birth_sex' },
      { text: 'Mother Age', value: 'mother_age' },
      { text: 'Birth Count', value: 'birth_count' },
    ],
    data: [],
  }),
  computed: {
    isFilterOn() {
      return !!this.siteSelected || !!this.motherAgeSelected || !!this.birthOrderSelected || !!this.sexSelected
    },
    filterSelected() {
      return `${this.siteSelected}|${this.motherAgeSelected}|${this.birthOrderSelected}|${this.sexSelected}`
    },
    totalNumberCount() {
      return this.data.reduce((total, item) => total + Number(item.birth_count), 0)
    }
  },
  watch: {
    filterSelected: function () {
      this.clearData()
    }
  },
  methods: {
    ...mapMutations('notification', ['showNotification']),
    fetchData() {
      if (!this.isFilterOn) return

      this.data = this.filterData(mockData.result.records, this.siteSelected, this.birthOrderSelected, this.motherAgeSelected, this.sexSelected)

      this.showNotification({ message: 'Data Fetched', color: 'success' })
    },
    clearData(isButtonClick = false) {
      this.data = []
      if (isButtonClick) this.showNotification({ message: 'Data Cleared', color: 'warning' })
    },
    resetFilter() {
      this.siteSelected = null
      this.motherAgeSelected = null
      this.birthOrderSelected = null
      this.sexSelected = null
      this.showNotification({ message: 'Filter Reset', color: 'info' })
    },
    filterData(data, siteSelected, birthOrderSelected, motherAgeSelected, sexSelected) {
      return data
        .filter(item => siteSelected && siteSelected.length > 0 ? siteSelected.includes(item.site_id) : true)
        .filter(item => birthOrderSelected && birthOrderSelected.length > 0 ? birthOrderSelected.includes(item.birth_order) : true)
        .filter(item => motherAgeSelected && motherAgeSelected.length > 0 ?  motherAgeSelected.includes(item.mother_age) : true)
        .filter(item => sexSelected && sexSelected.length > 0 ? sexSelected.includes(item.birth_sex) : true)
        .filter(item => item.birth_count > 0)
    },
    getColor(count) {
      return count > 0 ? 'red' : ''
    }
  }
}
</script>