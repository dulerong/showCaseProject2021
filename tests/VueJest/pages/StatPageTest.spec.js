import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'

import { filterStatData } from '~js/plugins/statPageConstants'

import StatPage from '~pages/StatPage'
import mockData from '~js/mockData/stat'

const mockRecords = mockData.result.records

describe('Stat Page', () => {
  const localVue = createLocalVue()

  localVue.use(Vuex)
  localVue.use(VueI18n)

  let vuetify
  let i18n
  const i18nConfig = {
    locale: 'en'
  }
  let store
  let notification
  let notificationMutations

  beforeEach(() => {
    vuetify = new Vuetify()
    i18n = new VueI18n(i18nConfig)
    notificationMutations = {
      showNotification: jest.fn(),
    }
    notification = {
      namespaced: true,
      mutations: notificationMutations
    }
    store = new Vuex.Store({
      modules: {
        notification
      }
    })
  })

  afterEach(() => {    
    jest.clearAllMocks()
  })

  const mountFunction = options => {
    return mount(StatPage, {
      localVue,
      vuetify,
      store,
      i18n,
      mocks: {
        $t: (msg) => msg
      },
      ...options
    })
  }

  it('Page renders properly', () => {
    const wrapper = mountFunction()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Method fetchData: populates data property', async () => {
    const wrapper = mountFunction()
    const siteSelected = ['新北市板橋區']
    const sexSelected = ['男']
    const numberRecordsMatched = filterStatData(
      mockRecords,
      siteSelected,
      null,
      null,
      sexSelected
    ).length

    await wrapper.setData({ siteSelected, sexSelected })

    const button = wrapper.find('.v-btn')
    expect(button.text()).toEqual('stat.button.fetch')

    button.vm.$emit('click')
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.data.length).toEqual(numberRecordsMatched)
  })

  it('Method clearData: clears data to empty array', () => {
    const options = {
      data: () => ({
        data: mockRecords
      }),
    }
    const wrapper = mountFunction(options)
    expect(wrapper.vm.data.length).not.toEqual(0)

    wrapper.vm.clearData()
    expect(wrapper.vm.data.length).toEqual(0)
  })

  it('Method resetFilter: resets filter to null', () => {
    const options = {
      data: () => ({
        siteSelected: ['新北市板橋區'],
      })
    }
    const wrapper = mountFunction(options)
    expect(wrapper.vm.siteSelected).not.toEqual([])

    wrapper.vm.resetFilter()
    expect(wrapper.vm.siteSelected).toEqual([])
  })

  it('Computed property: totalNumberCount calculates correctly', () => {
    const siteSelected = ['新北市板橋區']
    const sexSelected = ['男']
    const numberRecordsMatched = filterStatData(
      mockRecords,
      siteSelected,
      null,
      null,
      sexSelected
    )

    const actualNumberCount = numberRecordsMatched.reduce((total, item) => {
      return total + Number(item.birth_count)
    }, 0)

    const localThis = { data: numberRecordsMatched }

    expect(StatPage.computed.totalNumberCount.call(localThis))
      .toEqual(actualNumberCount)
  })

  it('Computed property: isFilterOn works correctly', () => {
    const localThisWithFilterOff = {
      siteSelected: [],
      motherAgeSelected: [],
      birthOrderSelected: [],
      sexSelected: []
    }
    expect(StatPage.computed.isFilterOn.call(localThisWithFilterOff))
      .toBeFalsy()

    const localThisWithFilterOn = {
      ...localThisWithFilterOff,
      siteSelected: ['新北市板橋區']
    }
    expect(StatPage.computed.isFilterOn.call(localThisWithFilterOn))
      .toBeTruthy()
  })

  it('Watch locale: fetchData triggerd when locale changes', async () => {
    const mockFetchData = jest.spyOn(StatPage.methods, 'fetchData')
    const wrapper = mountFunction()

    wrapper.vm.$i18n.locale = 'en'
    await wrapper.vm.$nextTick()
    expect(mockFetchData).not.toHaveBeenCalled()

    wrapper.vm.$i18n.locale = 'ch'
    await wrapper.vm.$nextTick()
    expect(mockFetchData).toHaveBeenCalled()
  })

  it('Watch filterSelected: triggers clearData', async () => {
    const mockClearData = jest.spyOn(StatPage.methods, 'clearData')
    const wrapper = mountFunction()

    await wrapper.setData({ siteSelected: [] })
    const initialFilterSelected = wrapper.vm.filterSelected
    expect(mockClearData).not.toHaveBeenCalled()

    await wrapper.setData({ siteSelected: ['新北市板橋區'] })
    const currentFilterSelected = wrapper.vm.filterSelected
    expect(currentFilterSelected).not.toEqual(initialFilterSelected)
    expect(mockClearData).toHaveBeenCalled()
  })

  it('Store Mutation: showNotification called at correct time', async () => {
    const wrapper = mountFunction()
    const siteSelected = ['新北市板橋區']

    wrapper.vm.fetchData()

    expect(notificationMutations.showNotification).not.toHaveBeenCalled()

    await wrapper.setData({ siteSelected })
    wrapper.vm.fetchData()

    expect(notificationMutations.showNotification).toHaveBeenCalled()
  })
})