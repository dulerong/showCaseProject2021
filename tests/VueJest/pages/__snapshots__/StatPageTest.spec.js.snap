// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat Page Page renders properly 1`] = `
<div>
  <h1>stat.title</h1>
  <form novalidate="novalidate" class="v-form">
    <div class="row no-gutters">
      <div class="v-input theme--light v-text-field v-select v-select--chips v-select--is-multi">
        <div class="v-input__control">
          <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-7" class="v-input__slot">
            <div class="v-select__slot"><label for="input-7" class="v-label theme--light" style="left: 0px; position: absolute;">stat.form.site</label>
              <div class="v-select__selections"><input data-testid="statPageSelectDistrict" id="input-7" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
              <div class="v-input__append-inner">
                <div></div>
              </div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
              </div><input type="hidden">
            </div>
            <div class="v-menu">
              <!---->
            </div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row no-gutters">
      <div class="v-input theme--light v-text-field v-select v-select--chips v-select--is-multi">
        <div class="v-input__control">
          <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-13" class="v-input__slot">
            <div class="v-select__slot"><label for="input-13" class="v-label theme--light" style="left: 0px; position: absolute;">stat.form.motherAge</label>
              <div class="v-select__selections"><input data-testid="" id="input-13" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
              <div class="v-input__append-inner">
                <div></div>
              </div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
              </div><input type="hidden">
            </div>
            <div class="v-menu">
              <!---->
            </div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row no-gutters">
      <div class="v-input theme--light v-text-field v-select v-select--chips v-select--is-multi">
        <div class="v-input__control">
          <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-19" class="v-input__slot">
            <div class="v-select__slot"><label for="input-19" class="v-label theme--light" style="left: 0px; position: absolute;">stat.form.birthOrder</label>
              <div class="v-select__selections"><input data-testid="" id="input-19" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
              <div class="v-input__append-inner">
                <div></div>
              </div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
              </div><input type="hidden">
            </div>
            <div class="v-menu">
              <!---->
            </div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row no-gutters">
      <div style="width: 150px;">
        <div class="v-input theme--light v-text-field v-select v-select--chips v-select--is-multi">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-25" class="v-input__slot">
              <div class="v-select__slot"><label for="input-25" class="v-label theme--light" style="left: 0px; position: absolute;">stat.form.birthSex</label>
                <div class="v-select__selections"><input data-testid="" id="input-25" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
                <div class="v-input__append-inner">
                  <div></div>
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row justify-lg-start no-gutters justify-space-between"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default mr-2"><span class="v-btn__content">stat.button.fetch</span></button><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default info mx-2"><span class="v-btn__content">stat.button.reset</span></button><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default warning mx-2"><span class="v-btn__content">stat.button.clear</span></button></div>
  </form>
  <div class="row my-3 no-gutters">
    <h3>stat.result: 0 born</h3>
  </div>
  <div class="v-data-table v-data-table--has-bottom theme--light">
    <div class="v-data-table__wrapper">
      <table>
        <colgroup>
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
        </colgroup>
        <thead class="v-data-table-header">
          <tr>
            <th role="columnheader" scope="col" aria-label="stat.tableHeader.site" class="text-start"><span>stat.tableHeader.site</span></th>
            <th role="columnheader" scope="col" aria-label="stat.tableHeader.motherAge: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>stat.tableHeader.motherAge</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="stat.tableHeader.birthOrder: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>stat.tableHeader.birthOrder</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="stat.tableHeader.birthSex: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>stat.tableHeader.birthSex</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="stat.tableHeader.birthCount: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>stat.tableHeader.birthCount</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
          </tr>
        </thead>
        <tbody>
          <tr class="v-data-table__empty-wrapper">
            <td colspan="5">No data available</td>
          </tr>
        </tbody>
      </table>
    </div>
    <div class="v-data-footer">
      <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-42" class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="Rows per page:" id="input-42" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden" value="10">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-footer__pagination">â€“</div>
      <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
      <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
    </div>
  </div>
</div>
`;
